Wikipedia/Code/InsertMediaSearchResultsCollectionViewController.swift

```diff
@@ -246,39 +245,46 @@ extension InsertMediaSearchResultsCollectionViewController {
+            let imageURL = searchResult.imageURL(for: view.bounds.width)
         else {
             return nil
         }
-        previewingContext.sourceRect = view.convert(cell.bounds, from: cell)
-        return indexPath
+        let previewingViewController = InsertMediaSearchResultPreviewingViewController(imageURL: imageURL, searchResult: searchResult)
+        previewingViewController.apply(theme: theme)
+        return previewingViewController
     }
 
-    func previewingContext(_ previewingContext: UIViewControllerPreviewing, viewControllerForLocation location: CGPoint) -> UIViewController? {
-        guard
-            let indexPath = collectionViewIndexPathForPreviewingContext(previewingContext, location: location),
-            let searchResult = searchResults[safeIndex: indexPath.item],
-            let imageURL = searchResult.imageURL(for: view.bounds.width)
-        else {
+    override func collectionView(_ collectionView: UICollectionView, contextMenuConfigurationForItemAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
+        guard let vc = viewController(for: indexPath) as? InsertMediaSearchResultPreviewingViewController else {
             return nil
         }
-        let previewingViewController = InsertMediaSearchResultPreviewingViewController(imageURL: imageURL, searchResult: searchResult)
-        previewingViewController.selectImageAction = {
-            self.collectionView.selectItem(at: indexPath, animated: true, scrollPosition: .centeredVertically)
-            self.delegate?.insertMediaSearchResultsCollectionViewControllerDidSelect(self, searchResult: searchResult)
+        let previewProvider: () -> UIViewController? = {
+            return vc
         }
-        previewingViewController.moreInformationAction = { url in
-            self.navigate(to: url, useSafari: true)
+        return UIContextMenuConfiguration(identifier: nil, previewProvider: previewProvider) { (suggestedActions) -> UIMenu? in
+            let selectImageAction = UIAction(title: WMFLocalizedString("insert-media-image-preview-select-image-action-title", value: "Select image", comment: "Title for preview action that results in image selection"), handler: { [weak self] (_) in
+                guard let self = self else {
+                    return
+                }
+                self.collectionView.selectItem(at: indexPath, animated: true, scrollPosition: .centeredVertically)
+                self.delegate?.insertMediaSearchResultsCollectionViewControllerDidSelect(self, searchResult: vc.searchResult)
+            })
+            let moreInformationAction = UIAction(title: WMFLocalizedString("insert-media-image-preview-more-information-action-title", value: "More information", comment: "Title for preview action that results in presenting more information"), handler: { [weak self] (_) in
+                guard let url = vc.searchResult.imageInfo?.filePageURL else {
```

Comment:
We're digging rather far to get this, do you mind making a computed property on InsertMediaSearchResultPreviewingViewController to return this url?

Reply:
Done